test/sequential/feeLock.API.feePayment.test.ts

```diff
@@ -0,0 +1,399 @@
+    sudo,
+    defaultCurrencyValue,
+    sudo
+  );
+});
+
+beforeEach(async () => {
+  const api = getApi();
+
+  [testUser1] = setupUsers();
+
+  await setupApi();
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(firstCurrency, testUser1, defaultCurrencyValue),
+    Assets.mintNative(sudo),
+    Sudo.sudoAs(
+      sudo,
+      Xyk.createPool(
+        MGA_ASSET_ID,
+        defaultPoolVolumeValue,
+        firstCurrency,
+        defaultPoolVolumeValue
+      )
+    )
+  );
+
+  feeLockMetadata = JSON.parse(
+    JSON.stringify(await api.query.feeLock.feeLockMetadata())
+  );
+
+  periodLength = feeLockMetadata.periodLength;
+  feeLockAmount = feeLockMetadata.feeLockAmount;
+
+  if (periodLength > 20) {
+    await updateFeeLockMetadata(
+      sudo,
+      new BN(10),
+      new BN(feeLockAmount),
+      null,
+      null
+    );
+
+    feeLockMetadata = JSON.parse(
+      JSON.stringify(await api.query.feeLock.feeLockMetadata())
+    );
+
+    periodLength = feeLockMetadata.periodLength;
+  }
+});
+
+test("gassless- GIVEN a feeLock configured (only Time and Amount ) WHEN the user swaps AND the user has not enough MGAs and has enough TURs THEN the extrinsic fails on submission", async () => {
+  await clearMgaFromWhitelisted();
+
+  await testUser1.addTURTokens(sudo);
+  await testUser1.addMGATokens(sudo, new BN(2));
+
+  await checkErrorSellAsset(
+    "1010: Invalid Transaction: Fee lock processing has failed either due to not enough funds to reserve or an unexpected error"
+  );
+});
+
+test("gassless- GIVEN a feeLock configured (only Time and Amount )  WHEN the user swaps AND the user does not have enough MGAs THEN the extrinsic fails on submission", async () => {
+  await clearMgaFromWhitelisted();
+
+  await testUser1.addMGATokens(sudo, new BN(2));
+
+  await checkErrorSellAsset(
+    "1010: Invalid Transaction: Fee lock processing has failed either due to not enough funds to reserve or an unexpected error"
+  );
+});
+
+test("gassless- Given a feeLock correctly configured (only Time and Amount ) WHEN the user swaps AND the user has enough MGAs THEN the extrinsic is correctly submitted", async () => {
+  await clearMgaFromWhitelisted();
+
+  await testUser1.addMGATokens(sudo);
+  testUser1.addAsset(MGA_ASSET_ID);
+
+  const sellAssetsValue = thresholdValue.add(new BN(5));
+
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+  await testUser1.sellAssets(MGA_ASSET_ID, firstCurrency, sellAssetsValue);
+  await testUser1.refreshAmounts(AssetWallet.AFTER);
+
+  const tokenFees = testUser1
+    .getAsset(MGA_ASSET_ID)
+    ?.amountAfter.reserved!.sub(
+      testUser1.getAsset(MGA_ASSET_ID)?.amountBefore.reserved!
+    );
+
+  const userMgaFees = testUser1
+    .getAsset(MGA_ASSET_ID)
+    ?.amountAfter.free!.sub(
+      testUser1.getAsset(MGA_ASSET_ID)?.amountBefore.free!
+    )
+    .add(new BN(sellAssetsValue));
+
+  expect(tokenFees).bnEqual(new BN(0));
+  expect(userMgaFees).bnEqual(new BN(0));
```

Comment:
Plz add an assert ensuring that the swap happened. Like the user has now more `fistcurrency`than before.

Reply:
Should not `.amountAfter.reserved!` be greather than 0 ? 
After the swaps, tokens must be frozen if MGA is not whitelisted.

Reply:
I made a swap gt `thresholdValue`,  that's why we don't block any token

Reply:
Added `firstCurrencyDeposit`